eth/downloader/statesync.go

```diff
@@ -274,16 +274,19 @@ func (s *stateSync) Cancel() error {
 // receive data from peers, rather those are buffered up in the downloader and
 // pushed here async. The reason is to decouple processing from data receipt
 // and timeouts.
-func (s *stateSync) loop() error {
+func (s *stateSync) loop() (err error) {
 	// Listen for new peer events to assign tasks to them
 	newPeer := make(chan *peerConnection, 1024)
 	peerSub := s.d.peers.SubscribeNewPeers(newPeer)
 	defer peerSub.Unsubscribe()
+	defer func() {
+		err = s.commit(true)
+	}()
```

Comment:
This will override any error being returned and most probably replace it with `nil`. You'd need to keep the old error intact and only override it if it was nil.